handlers/discount-expiry-notifier/src/handlers/getOldPaymentAmount.ts

```diff
@@ -54,15 +62,167 @@ export const handler = async (event: GetOldPaymentAmountInput) => {
 	}
 };
 
-const calculateTotalAmount = (records: InvoiceItemRecord[]) => {
+const handleTargetDateBeforeToday = async (
+	zuoraClient: ZuoraClient,
+	parsedEvent: GetOldPaymentAmountInput,
+	lastPaymentDateBeforeDiscountExpiry: string,
+) => {
+	const oldPaymentAmount = await getOldPaymentAmountWhenTargetDateIsBeforeToday(
+		zuoraClient,
+		parsedEvent.zuoraSubName,
+		lastPaymentDateBeforeDiscountExpiry,
+	);
+	console.log(
+		'oldPaymentAmount (target date is before today):',
+		oldPaymentAmount,
+	);
+
+	return {
+		...parsedEvent,
+		lastPaymentDateBeforeDiscountExpiry,
+		oldPaymentAmount,
+	};
+};
+
+const handleTargetDateAfterToday = async (
+	zuoraClient: ZuoraClient,
+	parsedEvent: GetOldPaymentAmountInput,
+	lastPaymentDateBeforeDiscountExpiry: string,
+) => {
+	const oldPaymentAmount = await getOldPaymentAmountWhenTargetDateIsAfterToday(
+		zuoraClient,
+		parsedEvent.zuoraSubName,
+		parsedEvent.billingAccountId,
+		lastPaymentDateBeforeDiscountExpiry,
+	);
+	console.log(
+		'oldPaymentAmount (target date is after today):',
+		oldPaymentAmount,
+	);
+
+	return {
+		...parsedEvent,
+		lastPaymentDateBeforeDiscountExpiry,
+		oldPaymentAmount,
+	};
+};
+
+//when target date is today, whether we use the query or billing preview depends on whether a bill run
+//has been run today. If a bill run has been run today, we use the query, otherwise we use the billing preview
```

Comment:
this comment could probably be made more accurate?

```suggestion
//when target date is today, whether we use the query or billing preview depends on whether a bill run
//has been run today. We try the billing-preview first for the target date, then use a query if nothing is return from the billing-preview
```

Reply:
Yes, thanks for pointing out. I've struggled a bit with the automated tests for this, specifically with mocking certain functions, and it was holding me up quite a bit, so I'm going to focus on functional testing for now and come back to it

Reply:
Good shout